docs/data/data-grid/tree-data/tree-data.md

```diff
@@ -165,6 +165,61 @@ const invalidRows = [{ path: ['X'] }, { path: ['Y'] }, { path: ['X', 'X'] }];
 
 :::
 
+## Drag-and-drop tree data reordering
+
+With row reordering, users can reorder tree data or move rows from one group to another.
+
+To enable this feature with tree data, pass the `rowReordering` prop to the Data Grid component.
+You also need to pass the `setTreeDataPath()` prop to revert the operation done by [`getTreeDataPath()`](/x/api/data-grid/data-grid-pro/#data-grid-pro-prop-getTreeDataPath) while building the tree as row reordering can change the path of the row.
+
+```tsx
+<DataGridPro
+  columns={columns}
+  rows={rows}
+  treeData
+  getTreeDataPath={getTreeDataPath}
+  setTreeDataPath={setTreeDataPath}
+  rowReordering
+/>
+```
+
+{{"demo": "TreeDataReordering.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Reordering persistance
+
+To sync the updated row order with an external store, depending on how you manage the order of rows in the external store, you can use either [`processRowUpdate()`](/x/api/data-grid/data-grid-pro/#data-grid-pro-prop-processRowUpdate) or the [`onRowOrderChange()`](/x/api/data-grid/data-grid-pro/#data-grid-pro-prop-onRowOrderChange) callback, or both.
+
+The `processRowUpdate()` callback is triggered whenever a row is updated, so when doing a cross-parent reorder operation, the `path` value gets updated based on the new position of the row(s) which result in one or more `processRowUpdate()` calls. If the `path` value is part of the row data, you can use the `processRowUpdate()` callback to update the `path` value.
+
+After the reorder operation is successfully completed, the `onRowOrderChange()` callback is triggered, which contains information about the new row re-ordering of the format [`GridRowOrderChangeParams`](/x/api/data-grid/grid-row-order-change-params/).
+
+```tsx
+<DataGridPro
+  // Capture row updates, can be multiple in one reorder operation, only triggered
+  // when one or more row updates (i.e. a cross parent reorder operation) are performed
+  processRowUpdate={processRowUpdate}
+  // Capture row re-ordering operations, always triggered once
+  // after the reorder operation is successfully completed
+  onRowOrderChange={handleRowOrderChange}
+/>
+```
+
+The demo below uses a custom data store binded using `useSyncExternalStore()` to persist the row data in the local storage.
+
+It maintains a row tree structure which is used to recompute path values after a reorder operation is performed, so a `processRowUpdate()` is not needed and only the `onRowOrderChange()` is sufficient for syncing the updated row order with the external store. This could be different with your specific implementation of row order persistence.
+
+:::success
+**Tip**: After some reorders, try refreshing the demo/page to see the persisted row order.
+:::
+
+{{"demo": "TreeDataSyncRowData.js", "bg": "inline", "defaultCodeOpen": false}}
+
+:::info
+The demo above uses the `isValidRowReorder()` prop to disable moving nodes "over" files (or leaf nodes) as it would convert them into parent groups.
```

Comment:
```suggestion
The demo above uses the `isValidRowReorder()` prop to disable moving nodes "over" files (or leaf nodes) because this would convert them into parent groups.
```